homeassistant/components/plaato/config_flow.py

```diff
@@ -1,10 +1,227 @@
+    PLACEHOLDER_WEBHOOK_URL,
 )
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__package__)
+
+
+class PlaatoConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handles a Plaato config flow."""
+
+    VERSION = 2
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize."""
+        self._init_info = {}
+        self._errors = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle user step."""
+
+        if user_input is not None:
+            self._init_info[CONF_DEVICE_TYPE] = PlaatoDeviceType(
+                user_input[CONF_DEVICE_TYPE]
+            )
+            self._init_info[CONF_DEVICE_NAME] = user_input[CONF_DEVICE_NAME]
+
+            return await self.async_step_api_method()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_DEVICE_NAME,
+                        default=self._init_info.get(CONF_DEVICE_NAME, None),
+                    ): str,
+                    vol.Required(
+                        CONF_DEVICE_TYPE,
+                        default=self._init_info.get(CONF_DEVICE_TYPE, None),
+                    ): vol.In(list(PlaatoDeviceType)),
+                }
+            ),
+            errors=self._errors,
+        )
+
+    async def async_step_api_method(self, user_input=None):
+        """Handle device type step."""
+
+        device_type = self._init_info[CONF_DEVICE_TYPE]
+
+        if user_input is None:
+            return await self._show_api_method_form(device_type)
+
+        token = user_input.get(CONF_TOKEN, None)
+        use_webhook = user_input.get(CONF_USE_WEBHOOK, False)
+
+        if not token and not use_webhook:
+            self._errors["base"] = PlaatoConfigFlow._get_error(device_type)
+            return await self._show_api_method_form(device_type)
+
+        self._init_info[CONF_USE_WEBHOOK] = use_webhook
+        self._init_info[CONF_TOKEN] = token
+        self._errors = {}
+        return await self.async_step_webhook()
+
+    async def async_step_webhook(self, user_input=None):
+        """Validate config step."""
+
+        use_webhook = self._init_info[CONF_USE_WEBHOOK]
+
+        if use_webhook and user_input is None:
+            webhook_id, webhook_url, cloudhook = await self._get_webhook_id()
+            self._init_info[CONF_WEBHOOK_ID] = webhook_id
+            self._init_info[CONF_CLOUDHOOK] = cloudhook
+
+            return self.async_show_form(
+                step_id="webhook",
+                errors=self._errors,
+                description_placeholders={
+                    PLACEHOLDER_WEBHOOK_URL: webhook_url,
+                    PLACEHOLDER_DOCS_URL: DOCS_URL,
+                },
+            )
+
+        return await self._async_step_create_entry()
+
+    async def _async_step_create_entry(self):
```

Comment:
```suggestion
    async def _async_create_entry(self):
```

This isn't a step as its not listed in `strings.json` and doesn't show a form